include/aspect/geometry_model/initial_topography_model/prm_polygon.h

```diff
@@ -0,0 +1,79 @@
+/*
+  Copyright (C) 2011, 2012, 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__geometry_model__initial_topography_prm_polygon_h
+#define __aspect__geometry_model__initial_topography_prm_polygon_h
+
+#include <aspect/geometry_model/initial_topography_model/interface.h>
+
+
+namespace aspect
+{
+  namespace InitialTopographyModel
+  {
+    using namespace dealii;
+
+    /**
+     * A class that describes a box geometry of certain width, height, and
+     * depth (in 3d).
+     */
+    template <int dim>
+    class PrmPolygon : public Interface<dim>
+    {
+      public:
+        /**
+         * Generate a coarse mesh for the geometry described by this class.
+         */
+        virtual
+        double value (const Point<dim-1> &p) const;
+
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+      private:
+        /**
+         * The values of the topography are stored in a vector.
+         */
+        std::vector<double> topography_values;
+
+        /**
+         * The polygons and their points are stored in this vector.
+         */
+        std::vector<std::vector<std::vector<double> > > point_lists;
```

Comment:
Since you know the size of the inner vector, can you change the inner vector to `Point<dim>` or `Point<dim-1>` or whatever is the appropriate type?